### DISCUSSION TITLE
Pydantic 2.0 and higher support

### DISCUSSION THREAD

**Discussion by HYBRID-BEING** [permitio/fastapi_websocket_rpc#23] [General]
Any plans to support more recent versions of pydantic? Pydantic enourages to migrate to 2.x version, however, atm, pydantic dependency for both rpc and pubsub packages is limited to versions between 1.9.1 and 2.

Not sure how many incompatibilities are there currently with higher versions, but from quick search it didn't look like pydantic is used **too** heavily.

In any case, pydantic's got a nice migration guide, explaining the changes and ways to migrate (including migration tool) https://docs.pydantic.dev/latest/migration/

**Comment by orweis** [comment]
Hi @HYBRID-BEING :) 
Yes we have such plans (As part of Permit.io's Roadmap to migrate most of our code to Pydantic 2- and the effort would also cover our OSS projects) - as the tickets haven't been assigned yet - I can only give you a range of a few months (but it will probably be sooner).

If you'd like to speed up the process by contributing a migration PR,w e'd be delighted to review/accept it - and also send you some swag ;) 

CC: @asafc , @shaulk , @roekatz , @omer9564
